tests/src/python/test_qgsvariantutils.py

```diff
@@ -43,6 +43,71 @@ def test_is_numeric_type(self):
         self.assertFalse(QgsVariantUtils.isNumericType(QMetaType.Type.QDate))
         self.assertFalse(QgsVariantUtils.isNumericType(QMetaType.Type.QTime))
 
+    def test_display_string(self):
+        # string
+        self.assertEqual(QgsVariantUtils.displayString(QVariant("foo bar")), "foo bar")
+        self.assertEqual(
+            QgsVariantUtils.displayString(QVariant("foo bar"), 2), "foo bar"
+        )
+
+        # integer
+        self.assertEqual(QgsVariantUtils.displayString(QVariant(23)), "23")
+        self.assertEqual(QgsVariantUtils.displayString(QVariant(23), 1), "23")
+
+        # float
+        self.assertEqual(
+            QgsVariantUtils.displayString(QVariant(3460.456)), "3\u202f460,456"
+        )
+        self.assertEqual(
+            QgsVariantUtils.displayString(QVariant(3460.456), 2), "3\u202f460,46"
+        )
+        self.assertEqual(
+            QgsVariantUtils.displayString(QVariant(3460.453), 2), "3\u202f460,45"
+        )
+        self.assertEqual(QgsVariantUtils.displayString(QVariant(-12.45)), "-12,45")
+        self.assertEqual(QgsVariantUtils.displayString(QVariant(-12.45), 1), "-12,4")
```

Comment:
If the function handles localized representation for double/float can you test for Fr_FR locale as the decimal separator is not the same?